knowledge-base/drawer-click-in-dropdown-closes-drawer.md

```diff
@@ -0,0 +1,136 @@
+---
+title: Click in dropdown closes the Drawer
+description: Click in dropdown closes the Drawer
+type: troubleshooting
+page_title: Click in dropdown closes the Drawer
+slug: drawer-kb-click-in-dropdown-closes-drawer
+position: 
+tags: drawer, dropdown, click, close
+ticketid: 1575100
+res_type: kb
+---
+
+## Environment
+
+<table>
+    <tr>
+        <td>Product</td>
+        <td>Drawer for Blazor</td>
+    </tr>
+    <tr>
+        <td>Product</td>
+        <td>AutoComplete for Blazor, ComboBox for Blazor, DropDownList for Blazor, MultiSelect for Blazor</td>
+    </tr>
+    <tr>
+        <td>Product</td>
+        <td>DatePicker for Blazor, DateRangePicker for Blazor, DateTimePicker for Blazor, DateTimePicker for Blazor</td>
+    </tr>
+</table>
+
+
+## Description
+
+I have a DropDownList inside a Drawer. The Drawer uses the Overlay mode. When I click in the dropdown to select an item, the Drawer closes. Click and drag the scrollbar in the dropdown also closes the Drawer.
+
+The same behavior occurs if I place a DatePicker inside a Drawer - click in the calendar popup closes the Drawer.
+
+How I can prevent this? How to keep the Drawer open when selecting items in the dropdown and/or calendar?
+
+## Cause\Possible Cause(s)
+
+By design, the popup elements of our components are not rendered in their place of declaration but on root level. Thus, when you are using a [DropDownList]({%slug components/dropdownlist/overview%}), for example, its popup is not technically part of the DropDownList component, it is rendered outside of it.
+
+As the popup of the DropDownList is rendered on root level, it is practically outside of the Drawer as well. So, when you click on an item in the dropdown, you are basically clicking outside of the Drawer. By design of its [Overlay mode]({%slug drawer-modes%}#overlay-mode), clicking outside of the Drawer container is expected to close the Drawer.
+
+You can verify this yourself using your dev tools to inspect the rendering. You can also test selecting an item from the DropDownList with keyboard navigation only (no click in the popup) - as a result, the Drawer will remain opened as you haven't clicked outside of it, in the DropDownList popup. The same behavior can be observed if you scroll without a click in the scrollbar.
+
+## Solution
+
+To handle the scenario, do one of the following:
```

Comment:
Provide the two options here and link via anchors. In this way, readers who immediately reject one option will see and navigate to the other one.